code/datums/gamemode/factions/faction.dm

```diff
@@ -35,11 +35,15 @@
 	var/datum/objective_holder/objective_holder
 	var/datum/role/roletype = /datum/role
 	var/logo_state = "synd-logo"
+	var/list/hud_icons = list()
 
 /datum/faction/New()
 	..()
 	objective_holder = new
 	objective_holder.faction = src
+	for (var/datum/faction/F in ticker.mode.factions)
```

Comment:
Have a list of factions from which you remove src and loop through that.
BYOND just does nothing when you ask to remove an item from a list and the list doesn't contain it.